In [1]:
import numpy as np
import math
from scipy.stats import norm
from scipy.stats import t

1.Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

считаем значение z для интервала 0.95

In [2]:
z=norm.ppf((1+0.95)/2)
z

1.959963984540054

считаем интервал

In [3]:
d=z*16/math.sqrt(256)
d

1.959963984540054

Ответ 1: Доверительный интервал: 80 +/- 1.959963984540054

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95. 

In [4]:
ar=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

считаем кол-во в выборке, ско и мат ожидание

In [5]:
n=ar.size
std=ar.std(ddof=1)
mean=ar.mean()
n,std,mean

(10, 0.4508017549014448, 6.590000000000001)

считаем интервал

In [6]:
z=norm.ppf((1+0.95)/2)
d=z*std/math.sqrt(n)
d

0.279404688242094

Ответ 2: Значение величины 6,59 +/-0.279404688242094

3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

определяем переменные α, выборка, дисперсия и ско

In [7]:
a=0.05
n=100
disp=4
std=math.sqrt(disp)

считаем критическое z для 5%

In [8]:
zt=norm.ppf(1-a)
zt

1.6448536269514722

считаем z для 17,5мм

In [9]:
zh=(17.5-17)/std/math.sqrt(n)
zh

0.025

сравним zt и zh, если zh меньше zt то H0 гипотеза верна на уровне значимости 5%

In [10]:
if zt > zh:
    print("Ответ 3: гипотеза H0 верна, шарики имеют средний диаметр 17 мм")

Ответ 3: гипотеза H0 верна, шарики имеют средний диаметр 17 мм


4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [11]:
ar=np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [12]:
H0=200
a=0.99
n=10

считаем ско и мат ожидание

In [13]:
std=ar.std(ddof=1)
mean=ar.mean()
std,mean

(4.453463071962462, 198.5)

считаем критическое z для 1%
т.к. 200>198.5 будем считать слева. Гипотеза H0 будет верна если zt<zh

In [14]:
zt=norm.ppf((1-a)/2)
zt

-2.5758293035489004

считаем z для 198.5г.

In [15]:
zh=(mean-H0)/std/math.sqrt(n)
zh

-0.10651074037450894

сравним zt и zh, если zh больше zt то H0 гипотеза верна на уровне значимости 99%

In [16]:
if zt < zh:
    print("Ответ 4: утверждение продавца верно")

Ответ 4: утверждение продавца верно
